packages/core/test/features/st-mixin.spec.ts

```diff
@@ -41,6 +41,62 @@ describe(`features/st-mixin`, () => {
 
         shouldReportNoDiagnostics(meta);
     });
+    it(`should append mixin declarations (within nesting)`, () => {
+        const { sheets } = testStylableCore(`
+            .mix {
+                propA: blue;
+                propB: green;
+            }
+
+            /* @rule .entry__empty {propA: blue; propB: green;} */
+            .empty {
+                -st-mixin: mix;
+            }
+
+            .insert {
+                /* @rule .entry__child {before:1; propA:blue; propB:green; after: 2} */
+                .child {
+                    before: 1;
+                    -st-mixin: mix;
+                    after: 2;
+                }
+            }
+        `);
+
+        const { meta } = sheets['/entry.st.css'];
+
+        shouldReportNoDiagnostics(meta);
+    });
+    it.skip(`should keep other nested rules`, () => {
```

Comment:
we need to make the testkit support nesting.

Reply:
What do you mean? - you can place `@rule <selector>` above each of the nested rules?

Reply:
I see now, the top checks both declarations and rules,  but the test-kit doesn't support it.